### PR TITLE
Add CMake options to the documentation

### PR DESCRIPTION
**Pull request by dangars**
I added several new CMake options to the documentation. Are there any missing new options?

Note that the options are not in strict alphabetic order. Logically `DEAL_II_PETSC_WITH_COMPLEX` should be after `DEAL_II_WITH_PETSC`, but the alphabetic order is then broken.
```
DEAL_II_WITH_P4EST
DEAL_II_WITH_PETSC
DEAL_II_PETSC_WITH_COMPLEX
DEAL_II_WITH_SCALAPACK
DEAL_II_WITH_SLEPC
```

### CODE REVIEW

File: doc/users/cmake.html [hunk 1]
```diff
@@ -470,6 +471,7 @@ <h3>Feature configuration</h3>
 DEAL_II_WITH_OPENCASCADE
 DEAL_II_WITH_P4EST
 DEAL_II_WITH_PETSC
+DEAL_II_PETSC_WITH_COMPLEX
```
Comment (masterleinad) [inline doc/users/cmake.html]: This is a feature that is detected while configuring. It should not be set by the user.

Comment (dangars) [reply doc/users/cmake.html]: That makes sense. I removed it :)

**Comment by masterleinad** [comment]
`GINKGO`, `GMSH` and `SYMENGINE` are missing.

**Comment by dangars** [comment]
I added the missing CMake options

**Comment by masterleinad** [comment]
Thanks!

**Pull request merged**